File: filter-openskidata.py
import json, sys

BBOX = [-110.92, 43.55, -110.80, 43.62]

def coord_in_bbox(c):
    return BBOX[0] <= c[0] <= BBOX[2] and BBOX[1] <= c[1] <= BBOX[3]

def any_coord_in_bbox(geom):
    t = geom.get('type','')
    coords = geom.get('coordinates',[])
    if t == 'Point': return coord_in_bbox(coords)
    if t == 'LineString': return any(coord_in_bbox(c) for c in coords)
    if t in ('MultiLineString','Polygon'): return any(coord_in_bbox(c) for ring in coords for c in ring)
    if t == 'MultiPolygon': return any(coord_in_bbox(c) for poly in coords for ring in poly for c in ring)
    return False

def process(inp, outp, label):
    print(f"Reading {label}...")
    with open(inp) as f:
        data = json.load(f)
    total = len(data['features'])
    print(f"Total: {total}")
    filtered = [ft for ft in data['features'] if ft.get('geometry') and any_coord_in_bbox(ft['geometry'])]
    print(f"JH bbox: {len(filtered)}")
    if filtered:
        print(f"\nSample props:\n{json.dumps(filtered[0].get('properties',{}), indent=2)[:2000]}")
    with open(outp, 'w') as f:
        json.dump({'type':'FeatureCollection','features':filtered}, f)
    print(f"Wrote {outp}\n")

process('/tmp/openskidata-runs-full.geojson', 'data/jh-runs.geojson', 'runs')
process('/tmp/openskidata-lifts-full.geojson', 'data/jh-lifts.geojson', 'lifts')

File: filter-openskidata.js
const fs = require('fs');

const BBOX = [-110.92, 43.55, -110.80, 43.62];

function coordInBBox(coord) {
  const [lng, lat] = coord;
  return lng >= BBOX[0] && lng <= BBOX[2] && lat >= BBOX[1] && lat <= BBOX[3];
}

function featureInBBox(feature) {
  const geom = feature.geometry;
  if (!geom || !geom.coordinates) return false;
  
  function checkCoords(coords, type) {
    if (type === 'Point') return coordInBBox(coords);
    if (type === 'LineString') return coords.some(c => coordInBBox(c));
    if (type === 'MultiLineString' || type === 'Polygon') return coords.some(ring => ring.some(c => coordInBBox(c)));
    if (type === 'MultiPolygon') return coords.some(poly => poly.some(ring => ring.some(c => coordInBBox(c))));
    return false;
  }
  
  return checkCoords(geom.coordinates, geom.type);
}

function filterFile(inputPath, outputPath, label) {
  console.log(`Reading ${label}...`);
  const raw = fs.readFileSync(inputPath, 'utf-8');
  const data = JSON.parse(raw);
  console.log(`Total features: ${data.features.length}`);
  
  const filtered = data.features.filter(featureInBBox);
  console.log(`Filtered to JH bbox: ${filtered.length} features`);
  
  // Show sample properties
  if (filtered.length > 0) {
    console.log(`\nSample properties (first feature):`);
    console.log(JSON.stringify(filtered[0].properties, null, 2));
  }
  
  const output = { type: 'FeatureCollection', features: filtered };
  fs.writeFileSync(outputPath, JSON.stringify(output));
  console.log(`Wrote ${outputPath}\n`);
}

filterFile('/tmp/openskidata-runs-full.geojson', 'data/jh-runs.geojson', 'runs');
filterFile('/tmp/openskidata-lifts-full.geojson', 'data/jh-lifts.geojson', 'lifts');
